unit-test-coverage/pc-rtems/src/coveragetest-cfe-psp-memory.c

```diff
@@ -0,0 +1,144 @@
+
+void Test_CFE_PSP_GetCDSSize(void)
+{
+    uint32 SizeOfCDS;
+
+    /* Test NULL pointer guard */
+    UtAssert_INT32_EQ(CFE_PSP_GetCDSSize(NULL), OS_ERROR);
+
+    /* Nominal/success path */
+    UtAssert_INT32_EQ(CFE_PSP_GetCDSSize(&SizeOfCDS), OS_SUCCESS);
+}
+
+void Test_CFE_PSP_WriteToCDS(void)
+{
+    uint32 CDSOffset = 8;
+    uint32 NumBytes  = 8;
+
+    /* Test NULL pointer guard */
+    UtAssert_INT32_EQ(CFE_PSP_WriteToCDS(NULL, CDSOffset, NumBytes), OS_ERROR);
+}
+
+void Test_CFE_PSP_ReadFromCDS(void)
+{
+    uint32 CDSOffset = 8;
+    uint32 NumBytes  = 8;
+
+    /* Test NULL pointer guard */
+    UtAssert_INT32_EQ(CFE_PSP_ReadFromCDS(NULL, CDSOffset, NumBytes), OS_ERROR);
+}
+
+void Test_CFE_PSP_GetResetArea(void)
+{
+    cpuaddr PtrToResetArea;
+    uint32  SizeOfResetArea;
+
+    /* Test both NULL pointer guards */
+    UtAssert_INT32_EQ(CFE_PSP_GetResetArea(NULL, &SizeOfResetArea), OS_ERROR);
+    UtAssert_INT32_EQ(CFE_PSP_GetResetArea(&PtrToResetArea, NULL), OS_ERROR);
+
+    /* Nominal/success path */
+    UtAssert_INT32_EQ(CFE_PSP_GetResetArea(&PtrToResetArea, &SizeOfResetArea), OS_SUCCESS);
+}
+
+void Test_CFE_PSP_GetUserReservedArea(void)
+{
+    cpuaddr PtrToUserArea;
+    uint32  SizeOfUserArea;
+
+    /* Test both NULL pointer guards */
+    UtAssert_INT32_EQ(CFE_PSP_GetUserReservedArea(NULL, &SizeOfUserArea), OS_ERROR);
+    UtAssert_INT32_EQ(CFE_PSP_GetUserReservedArea(&PtrToUserArea, NULL), OS_ERROR);
+
+    /* Nominal/success path */
+    UtAssert_INT32_EQ(CFE_PSP_GetUserReservedArea(&PtrToUserArea, &SizeOfUserArea), OS_SUCCESS);
+}
+
+void Test_CFE_PSP_GetVolatileDiskMem(void)
+{
+    cpuaddr PtrToVolDisk;
+    uint32  SizeOfVolDisk;
+
+    /* Test both NULL pointer guards */
+    UtAssert_INT32_EQ(CFE_PSP_GetVolatileDiskMem(NULL, &SizeOfVolDisk), OS_ERROR);
+    UtAssert_INT32_EQ(CFE_PSP_GetVolatileDiskMem(&PtrToVolDisk, NULL), OS_ERROR);
+
+    /* Nominal/success path */
+    UtAssert_INT32_EQ(CFE_PSP_GetVolatileDiskMem(&PtrToVolDisk, &SizeOfVolDisk), OS_SUCCESS);
+}
+
+void Test_CFE_PSP_SetupReservedMemoryMap(void)
+{
+    /* Test NULL ReservedMemBlock.BlockPtr path and early abort due to failure of malloc with a too-small heap */
+    UT_SetDataBuffer(UT_KEY(PCS_malloc), UT_TooSmallMemoryMapBuffer, sizeof(UT_TooSmallMemoryMapBuffer), false);
+    CFE_PSP_SetupReservedMemoryMap();
+    UtAssert_STUB_COUNT(PCS_abort, 1);
+    UT_ResetState(UT_KEY(PCS_malloc)); /* Reset so cleared for next test */
+    UT_ResetState(UT_KEY(OS_printf));  /* Reset so cleared for next test */
+
+    /* Nominal/success path with sufficiently sized 'heap' for emulated call to malloc */
+    UT_SetDataBuffer(UT_KEY(PCS_malloc), UT_MemoryMapBuffer, sizeof(UT_MemoryMapBuffer), false);
+    CFE_PSP_SetupReservedMemoryMap();
+    UtAssert_STUB_COUNT(OS_printf, 3); /* 3 calls to OS_printf along nominal path in CFE_PSP_SetupReservedMemoryMap */
+    UT_ResetState(UT_KEY(PCS_malloc)); /* Reset so cleared for future tests that invoke malloc */
```

Comment:
`UT_ResetState` at end of test shouldn't be necessary, `Psp_Test_Setup` resets everything prior to each test.